### DISCUSSION TITLE
beginner Build problem

### DISCUSSION THREAD

**Discussion by DrorSegev** [stella-cv/stella_vslam#278] [Q&A] [answered]
Hi,
When building stella_vslam I'm getting the following error:

home/dor/ddd/stella_vslam/src/stella_vslam/data/marker.cc:10:26: error: ‘mutex’ is not a member of ‘std’
   10 |     std::lock_guard<std::mutex> lock(mtx_position_);
      |                          ^~~~~

I have this alias defined: g++='g++ --std=c++11'
My gcc version is 9.4.0

Thanks

**Comment by DrorSegev** [comment]
Update:
I've added #include <mutex> in marker.h
Now the next build error is:
home/dor/ddd/stella_vslam/src/stella_vslam/marker_detector/base.h:27:21: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
   27 |          const std::shared_ptr<marker_model::base>& marker_model,
      |  
Will Continue update

**Comment by DrorSegev** [comment]
I've added #include <memory> in marker_detector/base.h
and build OK

**Comment by ymd-stella** [comment, marked as answer]
Please create a pull request if you would like to do so.

**Reply by ymd-stella** [reply]
I have fixed it.

**Comment by ymd-stella** [comment]
@DrorSegev 
I estimate that depending on the version of OpenCV you are using, it will not cause errors. May I ask what version of OpenCV you are using so that I can test?

**Reply by DrorSegev** [reply]
opencv-3.4.0

**Reply by DrorSegev** [reply]
BTW - what format is the map-db file is stored?
Dr

**Reply by ymd-stella** [reply]
MessagePack or sqlite3. default is MessagePack.

**Comment by ymd-stella** [comment]
> When building stella_vslam I'm getting the following error:
> 
> home/dor/ddd/stella_vslam/src/stella_vslam/data/marker.cc:10:26: error: ‘mutex’ is not a member of ‘std’
> 10 | std::lock_guardstd::mutex lock(mtx_position_);
> | ^~~~~

In my environment, I was also getting the following error

Is it the same in your environment?

```shell-session
In file included from /openvslam/src/stella_vslam/data/marker.cc:1:
/openvslam/src/stella_vslam/data/marker.h:37:18: error: 'mutex' in namespace 'std' does not name a type
   37 |     mutable std::mutex mtx_position_;
      |                  ^~~~~
```

**Reply by AMRitecs** [reply]
In my case, the compile error was solved by adding missing headers to the following files:
stella_vslam/data/marker.h 
#include < thread>
#include < mutex>

stella_vslam/marker_detector/base,h
#include < memory>
